packages/@lwc/engine-server/src/apis/render-component.ts

```diff
@@ -50,6 +51,13 @@ export function renderComponent(
         );
     }
 
+    if (!isObject(options) || isNull(options)) {
+        throw new TypeError(
+            `"renderComponent" expects an object as the fourth parameter but instead received ${options}.`
+        );
+    }
+
+    const renderer = new SsrRenderer(options);
```

Comment:
Let's normalize `options` before passing it to `SsrRender` and `serializeElement`. This would avoid having to deal with the `undefined` cases in both.